test/attachment-reference.spec.js

```diff
@@ -184,16 +184,72 @@ describe('Attachment reference validation type', () => {
           _attachments: {
             'foo.bar': { length: 151 }
           },
-          type: 'attachmentsDoc',
+          type: 'attachmentReferencesDoc',
           dynamicMaxSizeValidationProp: 'foo.bar',
           dynamicMaxSize: 150
         };
 
         testFixture.verifyDocumentNotCreated(
           doc,
-          'attachmentsDoc',
+          'attachmentReferencesDoc',
           errorFormatter.maximumSizeAttachmentViolation('dynamicMaxSizeValidationProp', 150));
       });
     });
   });
+
+  describe('regular expression pattern constraint', () => {
+    describe('with static validation', () => {
+      it('allows an attachment whose name matches the pattern', () => {
+        const doc = {
+          _id: 'my-doc',
+          type: 'attachmentReferencesDoc',
+          staticRegexPatternValidationProp: 'a03.hc'
+        };
+
+        testFixture.verifyDocumentCreated(doc);
+      });
+
+      it('allows an attachment whose name violates the pattern', () => {
+        const doc = {
+          _id: 'my-doc',
+          type: 'attachmentReferencesDoc',
+          staticRegexPatternValidationProp: '123ABC'
+        };
+
+        testFixture.verifyDocumentNotCreated(
+          doc,
+          'attachmentReferencesDoc',
+          errorFormatter.attachmentReferenceRegexPatternViolation('staticRegexPatternValidationProp', /^[a-z][a-z0-9]*\.[a-z]+$/));
+      });
+    });
+
+    describe('with dynamic validation', () => {
+      const expectedRegexString = '^\\d+\\.[a-z]+$';
+
+      it('allows an attachment whose name matches the pattern', () => {
+        const doc = {
+          _id: 'my-doc',
+          type: 'attachmentReferencesDoc',
+          dynamicRegexPattern: expectedRegexString,
+          dynamicRegexPatternValidationProp: '66134.txt'
+        };
+
+        testFixture.verifyDocumentCreated(doc);
+      });
+
+      it('allows an attachment whose name violates the pattern', () => {
```

Comment:
The name of this spec does not align with it's purposea, applies to the static case above as well.

Reply:
Oops. I'll fix that.